### PR TITLE
Update Scan api to return Long instead of String

### LINKED ISSUE #208: Scan functions have inconsistent API
The Scan functions, SSCAN for example, take the cursor as a Long but the subsequent cursor in the return value is a String

```
final def sScan(
    key: String,
    cursor: Long,
```

Returns `ZIO[RedisExecutor, RedisError, (String, Chunk[String])]`

This creates a burden on the user to (safely) convert the cursor to a Long for the next iteration. For example:

```
 testM("iterate over multiple scans") {
          for {
            key                <- uuid
            _                  <- sAdd(key, "a", "b", "c", "d", "e")
            scan               <- sScan(key, 0L, count = Some(Count(3L)))
            (cursor, members)   = scan
            scan2              <- sScan(key, cursor.toLong, count = Some(Count(3L)))
            (cursor2, members2) = scan2
          } yield assert(cursor)(isNonEmptyString) &&
            assert(members)(isNonEmpty) &&
            assert(cursor2)(equalTo("0"))
        }
```

### PR DESCRIPTION
**Pull request by justinhj**
fixes #208 

Updated API to return a Long
Updated response handling to convert the cursor from Redis to Long
Updated tests

### CODE REVIEW

File: redis/src/test/scala/zio/redis/SetsSpec.scala [hunk 1]
```diff
@@ -738,15 +738,15 @@ trait SetsSpec extends BaseSpec {
             _                <- sAdd(key, "a", "b", "c")
             scan             <- sScan(key, 0L)
             (cursor, members) = scan
-          } yield assert(cursor)(isNonEmptyString) &&
+          } yield assert(cursor)(equalTo(0L)) &&
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SetsSpec.scala:741]:
```suggestion
          } yield assert(cursor)(isZero) &&
```

File: redis/src/test/scala/zio/redis/SetsSpec.scala [hunk 2]
```diff
@@ -738,15 +738,15 @@ trait SetsSpec extends BaseSpec {
...
             assert(members)(isNonEmpty)
         },
         testM("empty set") {
           for {
             key              <- uuid
             scan             <- sScan(key, 0L)
             (cursor, members) = scan
-          } yield assert(cursor)(equalTo("0")) &&
+          } yield assert(cursor)(equalTo(0L)) &&
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SetsSpec.scala:749]:
```suggestion
          } yield assert(cursor)(isZero) &&
```

File: redis/src/test/scala/zio/redis/SetsSpec.scala [hunk 3]
```diff
@@ -755,16 +755,28 @@ trait SetsSpec extends BaseSpec {
             _                <- sAdd(key, "one", "two", "three")
             scan             <- sScan(key, 0L, Some("t[a-z]*".r))
             (cursor, members) = scan
-          } yield assert(cursor)(isNonEmptyString) &&
+          } yield assert(cursor)(equalTo(0L)) &&
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SetsSpec.scala:758]:
```suggestion
          } yield assert(cursor)(isZero) &&
```

**Review by anovakovic01 — APPROVED** [review]

File: redis/src/test/scala/zio/redis/SetsSpec.scala [hunk 4]
```diff
@@ -755,16 +755,28 @@ trait SetsSpec extends BaseSpec {
...
+          for {
+            key                <- uuid
+            _                  <- sAdd(key, "a", "b", "c", "d", "e")
+            scan               <- sScan(key, 0L, count = Some(Count(3L)))
+            (cursor, members)   = scan
+            scan2              <- sScan(key, cursor, count = Some(Count(3L)))
+            (cursor2, members2) = scan2
+          } yield assert(cursor)(isGreaterThan(0L)) &&
+            assert(members)(isNonEmpty) &&
+            assert(cursor2)(equalTo(0L))
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SetsSpec.scala:771]:
```suggestion
            assert(cursor2)(isZero)
```

File: redis/src/test/scala/zio/redis/SortedSetsSpec.scala [hunk 5]
```diff
@@ -794,22 +794,22 @@ trait SortedSetsSpec extends BaseSpec {
             _                <- zAdd(key)(MemberScore(1d, "atest"), MemberScore(2d, "btest"), MemberScore(3d, "ctest"))
             scan             <- zScan(key, 0L)
             (cursor, members) = scan
-          } yield assert(cursor)(isNonEmptyString) && assert(members)(isNonEmpty)
+          } yield assert(cursor)(equalTo(0L)) && assert(members)(isNonEmpty)
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SortedSetsSpec.scala:797]:
```suggestion
          } yield assert(cursor)(isZero) && assert(members)(isNonEmpty)
```

File: redis/src/test/scala/zio/redis/SortedSetsSpec.scala [hunk 6]
```diff
@@ -794,22 +794,22 @@ trait SortedSetsSpec extends BaseSpec {
...
         },
         testM("empty set") {
           for {
             key              <- uuid
             scan             <- zScan(key, 0L)
             (cursor, members) = scan
-          } yield assert(cursor)(equalTo("0")) && assert(members)(isEmpty)
+          } yield assert(cursor)(equalTo(0L)) && assert(members)(isEmpty)
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SortedSetsSpec.scala:804]:
```suggestion
          } yield assert(cursor)(isZero) && assert(members)(isEmpty)
```

File: redis/src/test/scala/zio/redis/SortedSetsSpec.scala [hunk 7]
```diff
@@ -794,22 +794,22 @@ trait SortedSetsSpec extends BaseSpec {
...
         },
         testM("with match over non-empty set") {
           for {
             key              <- uuid
             _                <- zAdd(key)(MemberScore(1d, "one"), MemberScore(2d, "two"), MemberScore(3d, "three"))
             scan             <- zScan(key, 0L, Some("t[a-z]*".r))
             (cursor, members) = scan
-          } yield assert(cursor)(isNonEmptyString) && assert(members)(isNonEmpty)
+          } yield assert(cursor)(equalTo(0L)) && assert(members)(isNonEmpty)
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SortedSetsSpec.scala:812]:
```suggestion
          } yield assert(cursor)(isZero) && assert(members)(isNonEmpty)
```

File: redis/src/test/scala/zio/redis/SortedSetsSpec.scala [hunk 8]
```diff
@@ -823,7 +823,7 @@ trait SortedSetsSpec extends BaseSpec {
                  )
             scan             <- zScan(key, 0L, None, Some(Count(3L)))
             (cursor, members) = scan
-          } yield assert(cursor)(isNonEmptyString) && assert(members)(isNonEmpty)
+          } yield assert(cursor)(equalTo(0L)) && assert(members)(isNonEmpty)
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SortedSetsSpec.scala:826]:
```suggestion
          } yield assert(cursor)(isZero) && assert(members)(isNonEmpty)
```

File: redis/src/test/scala/zio/redis/SortedSetsSpec.scala [hunk 9]
```diff
@@ -837,7 +837,7 @@ trait SortedSetsSpec extends BaseSpec {
                  )
             scan             <- zScan(key, 0L, Some("t[a-z]*".r), Some(Count(3L)))
             (cursor, members) = scan
-          } yield assert(cursor)(isNonEmptyString) && assert(members)(isNonEmpty)
+          } yield assert(cursor)(equalTo(0L)) && assert(members)(isNonEmpty)
```
Comment (mijicd) [inline redis/src/test/scala/zio/redis/SortedSetsSpec.scala:840]:
```suggestion
          } yield assert(cursor)(isZero) && assert(members)(isNonEmpty)
```

**Review by mijicd — CHANGES_REQUESTED** [review]
Just a few suggestions from my side, other than that it looks good to me.

**Comment by mijicd** [comment]
@justinhj I've committed the suggestions, let's wait for build to pass and it's good to go.

**Review by anovakovic01 — APPROVED** [review]

**Review by mijicd — APPROVED** [review]

**Pull request merged**